mirrord/agent/src/metrics.rs

```diff
@@ -0,0 +1,166 @@
+use std::{net::SocketAddr, sync::LazyLock};
+
+use axum::{routing::get, Router};
+use http::StatusCode;
+use prometheus::{register_int_gauge, IntGauge};
+use tokio::net::TcpListener;
+use tokio_util::sync::CancellationToken;
+use tracing::Level;
+
+use crate::error::AgentError;
+
+pub(crate) static OPEN_FD_COUNT: LazyLock<IntGauge> = LazyLock::new(|| {
+    register_int_gauge!(
+        "mirrord_agent_open_fd_count",
+        "amount of open file descriptors in mirrord-agent file manager"
+    )
+    .expect("Valid at initialization!")
+});
+
+pub(crate) static MIRROR_PORT_SUBSCRIPTION: LazyLock<IntGauge> = LazyLock::new(|| {
+    register_int_gauge!(
+        "mirrord_agent_mirror_port_subscription_count",
+        "amount of mirror port subscriptions in mirror-agent"
+    )
+    .expect("Valid at initialization")
+});
+
+pub(crate) static MIRROR_CONNECTION_SUBSCRIPTION: LazyLock<IntGauge> = LazyLock::new(|| {
+    register_int_gauge!(
+        "mirrord_agent_mirror_connection_subscription_count",
+        "amount of connections in mirror mode in mirrord-agent"
+    )
+    .expect("Valid at initialization!")
+});
+
+pub(crate) static STEAL_FILTERED_PORT_SUBSCRIPTION: LazyLock<IntGauge> = LazyLock::new(|| {
+    register_int_gauge!(
+        "mirrord_agent_steal_filtered_port_subscription_count",
+        "amount of filtered steal port subscriptions in mirrord-agent"
+    )
+    .expect("Valid at initialization!")
+});
+
+pub(crate) static STEAL_UNFILTERED_PORT_SUBSCRIPTION: LazyLock<IntGauge> = LazyLock::new(|| {
+    register_int_gauge!(
+        "mirrord_agent_steal_unfiltered_port_subscription_count",
+        "amount of unfiltered steal port subscriptions in mirrord-agent"
+    )
+    .expect("Valid at initialization!")
+});
+
+pub(crate) static STEAL_FILTERED_CONNECTION_SUBSCRIPTION: LazyLock<IntGauge> =
+    LazyLock::new(|| {
+        register_int_gauge!(
+            "mirrord_agent_steal_connection_subscription_count",
+            "amount of filtered connections in steal mode in mirrord-agent"
+        )
+        .expect("Valid at initialization!")
+    });
+
+pub(crate) static STEAL_UNFILTERED_CONNECTION_SUBSCRIPTION: LazyLock<IntGauge> =
```

Comment:
The way you manipulate this metric is hard to follow and looks very prone to errors :V
There's a lot of branching and early exits in the code.

I suggest you just:
1. Add `UnfilteredStealTask::new`
2. Inc the gauge there
3. Dec the gauge in `impl Drop for UnfilteredStealTask`